stl/src/vector_algorithms.cpp

```diff
@@ -523,9 +523,199 @@ __declspec(noalias) void __stdcall __std_rotate(void* _First, void* const _Mid,
+                    const uint8x16_t _Left2  = vld1q_u8(static_cast<uint8_t*>(_First) + 16);
+                    const uint8x16_t _Right1 = vld1q_u8(static_cast<uint8_t*>(_Last) + 0);
+                    const uint8x16_t _Right2 = vld1q_u8(static_cast<uint8_t*>(_Last) + 16);
+
+                    const uint8x16_t _Left1_reversed  = _Traits::_Rev(_Left1);
+                    const uint8x16_t _Left2_reversed  = _Traits::_Rev(_Left2);
+                    const uint8x16_t _Right1_reversed = _Traits::_Rev(_Right1);
+                    const uint8x16_t _Right2_reversed = _Traits::_Rev(_Right2);
+
+                    vst1q_u8(static_cast<uint8_t*>(_First) + 0, _Right2_reversed);
+                    vst1q_u8(static_cast<uint8_t*>(_First) + 16, _Right1_reversed);
+                    vst1q_u8(static_cast<uint8_t*>(_Last) + 0, _Left2_reversed);
+                    vst1q_u8(static_cast<uint8_t*>(_Last) + 16, _Left1_reversed);
+
+                    _Advance_bytes(_First, 32);
+                } while (_First != _Stop_at);
+            }
+
+            if (const size_t _Length = _Byte_length(_First, _Last); _Length >= 32) {
+                _Advance_bytes(_Last, -16);
+                const uint8x16_t _Left  = vld1q_u8(static_cast<uint8_t*>(_First));
+                const uint8x16_t _Right = vld1q_u8(static_cast<uint8_t*>(_Last));
+
+                const uint8x16_t _Left_reversed  = _Traits::_Rev(_Left);
+                const uint8x16_t _Right_reversed = _Traits::_Rev(_Right);
+
+                vst1q_u8(static_cast<uint8_t*>(_First), _Right_reversed);
+                vst1q_u8(static_cast<uint8_t*>(_Last), _Left_reversed);
+                _Advance_bytes(_First, 16);
+            }
+
+            if (const size_t _Length = _Byte_length(_First, _Last); _Length >= 16) {
+                _Advance_bytes(_Last, -8);
+                const uint8x8_t _Left  = vld1_u8(static_cast<uint8_t*>(_First));
+                const uint8x8_t _Right = vld1_u8(static_cast<uint8_t*>(_Last));
+
+                const uint8x8_t _Left_reversed  = _Traits::_Rev(_Left);
+                const uint8x8_t _Right_reversed = _Traits::_Rev(_Right);
+
+                vst1_u8(static_cast<uint8_t*>(_First), _Right_reversed);
+                vst1_u8(static_cast<uint8_t*>(_Last), _Left_reversed);
+                _Advance_bytes(_First, 8);
+            }
+
+            if (const size_t _Length = _Byte_length(_First, _Last); _Length >= 8) {
+                _Advance_bytes(_Last, -8);
+                const uint8x8_t _Left  = vld1_u8(static_cast<uint8_t*>(_First));
+                const uint8x8_t _Right = vld1_u8(static_cast<uint8_t*>(_Last));
+
+                const uint8x8_t _Left_reversed  = _Traits::_Rev(_Left);
+                const uint8x8_t _Right_reversed = _Traits::_Rev(_Right);
+
+                vst1_u8(static_cast<uint8_t*>(_First), _Right_reversed);
+                vst1_u8(static_cast<uint8_t*>(_Last), _Left_reversed);
+                _Advance_bytes(_First, 8);

```

Comment:
Likely to be eliminated by the compiler, but still looks pointless.
Unless if you want to maintain consistency with other occurrences, bat that consistency is a bit misleading.